core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -103,10 +125,72 @@ impl Cast {
             (DataType::LargeUtf8, DataType::Boolean) => {
                 Self::spark_cast_utf8_to_boolean::<i64>(&array, self.eval_mode)?
             }
-            _ => cast_with_options(&array, to_type, &CAST_OPTIONS)?,
+            (
+                DataType::Utf8,
```

Comment:
Don't we need to handle LargeUtf8?

Reply:
I have added this, but there are no tests for it. Do we have a way to test this?

Reply:
As far as I know, we don't have a way to have LargeUtf8 to test in Scala tests. If possible, maybe just unit test the function. If no, I think the logic is same as Utf8 so it should be fine to test Utf8 path only.